### PR TITLE
Workout Plan Generation & Multiplier Endpoints

### PR DESCRIPTION
**Pull request by zfrantzen**
Note that the database credentials will be removed from the codebase once we get the environment variables set up.

For some context on the code, the system currently supports only arm workouts (major_muscle_group_id = 1). However, we can easily support more in the future as all we have to do is add some data to the database and make a couple minor code changes.

Be careful when you use the generate plan endpoint as it will create quite a bit of data in the database. So please don't make a script that executes it over and over and over and over and over again for no reason as someone will have to clean up the mess it'll make.

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -38,4 +46,38 @@ app.delete('/users/:userID', cors(), function (req, res) {
...
+    JSON.parse(req.params.lengthMinutes), 
+    JSON.parse(req.params.targetMuscleGroup), 
+    dbConfig
+  );
+
+  res.json(result);
+})
+
+app.put('/users/:userId/workout-difficulty/increase/:factor&:targetMuscleGroup', cors(), async function (req, res) {
+  var result = await workout.modifyWorkoutDifficulty(
```
Comment (danaharlos) [inline server/server.js:62]: I like the way you made these functions,  I was having trouble putting my functions in a separate function call without them breaking.
I took some inspiration from you and now my api endpoints are much cleaner :)

File: server/src/data/workoutData.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        }
+    },
+
+    getUserMultiplier: function(targetMuscleGroup, multiplierId, dbConfig) {
+        try {
+            return sql
+              .connect(dbConfig)
+              .then((pool) => {
+                // Should probably be checking the database instead of hardcoding this,
+                // but it is okay for now... Will need to update db structure to fix properly
```
Comment (danaharlos) [inline server/src/data/workoutData.js:31]: good enough for M6 😄

File: server/src/data/workoutData.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    createWorkoutPlanEntry: function(userId, dbConfig) {
+        try {
+            return sql
+              .connect(dbConfig)
+              .then((pool) => {
+                return pool
+                  .request()
+                  .input("userId", sql.Int, userId)
+                  .query(
+                    "INSERT INTO workout_plan(user_profile_id) OUTPUT Inserted.id VALUES (@userId)"
```
Comment (danaharlos) [inline server/src/data/workoutData.js:61]: Curious if you've been able to test your insert statements, and if so - how?
I've only done the "create user" post statement, but still haven't tested

File: server/src/engine/workoutPlanGenerationEngine.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+const util = require('../util.js');
+const workoutData = require('../data/workoutData.js');
+
+const MAX_REST_SEC = 45;
+const MIN_REST_SEC = 15;
+const STD_REST_TIME_SEC = 30;
+const TIME_ESTIMATE_SHIFT_FACTOR = 0.5;
+
+module.exports = {
+    generateWorkoutPlan: function(lengthMinutes, possibleExercises, multiplier, planId) {
```
Comment (danaharlos) [inline server/src/engine/workoutPlanGenerationEngine.js:11]: This generate workout plan is already starting to look like complex logic to me. I wonder how much more complex we need to go to fill the project requirement

**Review by danaharlos — APPROVED** [review]
Looks great. I left a couple comments just to further my own understanding about the code

File: server/src/data/workoutData.js (hunk 3 above)
Comment (zfrantzen) [reply server/src/data/workoutData.js:61]: Yeah, I've been able to test all of the SQL queries. I just ran them and checked if the database changed as expected.

**Pull request merged**